src/devices/cpu/tlcs900/dasm900.cpp

```diff
@@ -1369,12 +1374,260 @@ const char *const tlcs900_disassembler::s_cond[16] =
+};
+
+tmp96c141_disassembler::tmp96c141_disassembler()
+	: tlcs900_disassembler(0x80, s_sfr_names)
+{
+}
+
+const char *const tmp96c141_disassembler::s_sfr_names[0x80] = {
+	/*00:*/ "P0",      "P1",      "P0CR",    "0x03",    "P1CR",    "P1FC",    "P2",      "P3",
+	/*08:*/ "P2CR",    "P2FC",    "P3CR",    "P3FC",    "P4",      "P5",      "P4CR",    "0x0f",
+	/*10:*/ "P4FC",    "0x11",    "P6",      "P7",      "P6CR",    "P7CR",    "P6FC",    "P7FC",
+	/*18:*/ "P8",      "P9",      "P8CR",    "P9CR",    "P8FC",    "P9FC",    "0x1e",    "0x1f",
+	/*20:*/ "TRUN",    "0x21",    "TREG0",   "TREG1",   "TMOD",    "TFFCR",   "TREG2",   "TREG3",
+	/*28:*/ "P0MOD",   "P1MOD",   "PFFCR",   "0x2b",    "0x2c",    "0x2d",    "0x2e",    "0x2f",
+	/*30:*/ "TREG4L",  "TREG4H",  "TREG5L",  "TREG5H",  "CAP1L",   "CAP1H",   "CAP2L",   "CAP2H",
+	/*38:*/ "T4MOD",   "T4FFCR",  "T45CR",   "0x3b",    "0x3c",    "0x3d",    "0x3e",    "0x3f",
+	/*40:*/ "TREG6L",  "TREG6H",  "TREG7L",  "TREG7H",  "CAP3L",   "CAP3H",   "CAP4L",   "CAP4H",
+	/*48:*/ "T5MOD",   "T5FFCR",  "0x4a",    "0x4b",    "PG0REG",  "PG1REG",  "PG01CR",  "0x4f",
+	/*50:*/ "SC0BUF",  "SC0CR",   "SC0MOD",  "BR0CR",   "SC1BUF",  "SC1CR",   "SC1MOD",  "BR1CR",
+	/*58:*/ "ODE",     "0x59",    "0x5a",    "0x5b",    "WDMOD",   "WDCR",    "ADMOD",   "0x5f",
+	/*60:*/ "ADREG0L", "ADREG0H", "ADREG1L", "ADREG1H", "ADREG2L", "ADREG2H", "ADREG3L", "ADREG3H",
+	/*68:*/ "B0CS",    "B1CS",    "B2CS",    "0x6b",    "0x6c",    "0x6d",    "0x6e",    "0x6f",
+	/*70:*/ "INTE0AD", "INTE45",  "INTE67",  "INTET10", "INTEPW10","INTET54", "INTET76", "INTES0",
+	/*78:*/ "INTES1",  "0x79",    "0x7a",    "IIMC",    "DMA0V",   "DMA1V",   "DMA2V",   "DMA3V"
+};
+
 
 u32 tlcs900_disassembler::opcode_alignment() const
 {
 	return 1;
 }
 
+void tlcs900_disassembler::decode_control_register_8(std::ostream &stream, uint8_t imm){
+	switch( imm )
+	{
+	case 0x22: util::stream_format(stream, " DMAM0"); break;
+	case 0x26: util::stream_format(stream, " DMAM1"); break;
+	case 0x2a: util::stream_format(stream, " DMAM2"); break;
+	case 0x2e: util::stream_format(stream, " DMAM3"); break;
+	default:
+		util::stream_format(stream, " unknown (encoding is 0x%02x)", imm);
+		break;
+	}
+}
+
+
+void tmp94c241_disassembler::decode_control_register_8(std::ostream &stream, uint8_t imm){
+	switch( imm )
+	{
+	case 0x42: util::stream_format(stream, " DMAM0"); break;
+	case 0x46: util::stream_format(stream, " DMAM1"); break;
+	case 0x4a: util::stream_format(stream, " DMAM2"); break;
+	case 0x4e: util::stream_format(stream, " DMAM3"); break;
+	case 0x52: util::stream_format(stream, " DMAM4"); break;
+	case 0x56: util::stream_format(stream, " DMAM5"); break;
+	case 0x5a: util::stream_format(stream, " DMAM6"); break;
+	case 0x5e: util::stream_format(stream, " DMAM7"); break;
+	default:
+		util::stream_format(stream, " unknown (encoding is 0x%02x)", imm);
+		break;
+	}
+}
```

Comment:
The control register definitions should be supplied as configuration on construction as well to avoid the need for these virtual member functions.  You can use (imm, name) pairs like the M6805 disassembler does for SFRs.

Also, the disassembly would probably be more readable if you just used the value for unknown/invalid values, rather than the extra descriptive text.